src/cascadia/TerminalControl/ControlInteractivity.h

```diff
@@ -0,0 +1,150 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+//
+// Module Name:
+// - ControlInteractivity.h
+//
+// Abstract:
+// - This is a wrapper for the `ControlCore`. It holds the logic for things like
+//   double-click, right click copy/paste, selection, etc. This is intended to
+//   be a UI framework-independent abstraction. The methods this layer exposes
+//   can be called the same from both the WinUI `TermControl` and the WPF
+//   control.
+//
+// Author:
+// - Mike Griese (zadjii-msft) 01-Apr-2021
+
+#pragma once
+
+#include "ControlInteractivity.g.h"
+#include "EventArgs.h"
+#include "../buffer/out/search.h"
+#include "cppwinrt_utils.h"
+
+#include "ControlCore.h"
+
+namespace Microsoft::Console::VirtualTerminal
+{
+    struct MouseButtonState;
+}
+
+namespace ControlUnitTests
+{
+    class ControlCoreTests;
+    class ControlInteractivityTests;
+};
+
+namespace winrt::Microsoft::Terminal::Control::implementation
+{
+    struct ControlInteractivity : ControlInteractivityT<ControlInteractivity>
+    {
+    public:
+        ControlInteractivity(IControlSettings settings,
+                             TerminalConnection::ITerminalConnection connection);
+
+        void GainFocus();
+        void UpdateSettings();
+        void Initialize();
+        winrt::com_ptr<ControlCore> GetCore();
+
+#pragma region InputMethods
+        void PointerPressed(const winrt::Windows::Foundation::Point mouseCursorPosition,

```

Comment:
file a todo for moving these to til :smile: (or, do it now because it is free!)

I know that this will not work immediately when it's over the boundary, but... the autoconverters can handle that (c++/winrt lets you implement a projection method using any compatible types!)

Reply:
and we're already not using IDL types.